core.go

```diff
@@ -122,3 +124,25 @@ type core struct {
 
 	fields map[string]interface{}
 }
+
+// follow same logic with sentry-go to filter unnecessary frames
+// ref:
+// https://github.com/getsentry/sentry-go/blob/362a80dcc41f9ad11c8df556104db3efa27a419e/stacktrace.go#L256-L280
+func filterFrames(frames []sentry.Frame) []sentry.Frame {
+	if len(frames) == 0 {
+		return nil
+	}
+	filteredFrames := make([]sentry.Frame, 0, len(frames))
+
+	for i := range frames {
+		// Skip zapsentry and zap internal frames, except for frames in _test packages (for
+		// testing).
```

Comment:
what's usecase not skipping test package?

Reply:
Thanks for asking 😄 
The use-case is: When we are executing the tests of `TheZeroSlave/zapsentry`, the frames won't be filtered.
It's for `TheZeroSlave/zapsentry` internal testing only.

TL;DR

Let's give it a try with [the example](https://github.com/TheZeroSlave/zapsentry/pull/14/files)

When:

```
if (strings.HasPrefix(frames[i].Module, "github.com/TheZeroSlave/zapsentry") ||
	strings.HasPrefix(frames[i].Function, "go.uber.org/zap")) &&
	!strings.HasSuffix(frames[i].Module, "_test") {
		continue
}
```

`recordedSentryEvent.Exception[0].Stacktrace` will be like this:

```
&{Frames:[
{Function:main Symbol: Module:main Package: Filename:_testmain.go AbsPath: Lineno:47 Colno:0 PreContext:[] ContextLine: PostContext:[] InApp:true Vars:map[]} 
{Function:ExampleAttachCoreToLogger Symbol: Module:github.com/TheZeroSlave/zapsentry_test Package: Filename: AbsPath:/.../projects/zapsentry/example_logger_test.go Lineno:84 Colno:0 PreContext:[		log.Fatal(err) 	} 	newLogger := zapsentry.AttachCoreToLogger(core, logger)  	// Send error log] ContextLine:	newLogger.Error("[error] something went wrong!", zap.String("method", "unknown")) PostContext:[ 	// Check output 	fmt.Println(recordedLogs.All()[0].Message) 	fmt.Println(recordedSentryEvent.Message) 	fmt.Println(recordedSentryEvent.Extra)] InApp:true Vars:map[]}] FramesOmitted:[]}
```

But if we change the logic to this:

```
if strings.HasPrefix(frames[i].Module, "github.com/TheZeroSlave/zapsentry") ||
	strings.HasPrefix(frames[i].Function, "go.uber.org/zap") {
	continue
}
```

And you can see there's no frame of `ExampleAttachCoreToLogger` in `recordedSentryEvent.Exception[0].Stacktrace` :

```
&{Frames:[
{Function:main Symbol: Module:main Package: Filename:_testmain.go AbsPath: Lineno:47 Colno:0 PreContext:[] ContextLine: PostContext:[] InApp:true Vars:map[]}] FramesOmitted:[]}
```